src/context/WalletContext.tsx

```diff
@@ -35,27 +33,23 @@ const WalletContextProvider = ({
     utxos,
   };
 
-  useEffect(() => {
-    if (!isWalletConnected) return;
-
-    ergo
-      .get_utxos()
-      .then((bs) => (bs ? bs.map((b) => fixErgoBox(b)) : bs))
-      .then((data: ErgoBox[] | undefined) => {
-        setUtxos(data ?? []);
-      });
-  }, [isWalletConnected]);
-
-  useInterval(() => {
-    if (!isWalletConnected) return;
-
-    ergo
-      .get_utxos()
-      .then((bs) => (bs ? bs.map((b) => fixErgoBox(b)) : bs))
-      .then((data: ErgoBox[] | undefined) => {
-        setUtxos(data ?? []);
-      });
-  }, 10 * 1000);
+  function fetchUtxos(): void {
+    if (isWalletConnected) {
+      ergo
+        .get_utxos()
+        .then((bs) => (bs ? bs.map((b) => ergoBoxFromProxy(b)) : bs))
+        .then((data: ErgoBox[] | undefined) => {
+          setUtxos(data ?? []);
+        });
+      return
+    } else {
+      return
+    }
+  }
+
+  useEffect(() => fetchUtxos(), [isWalletConnected]);
```

Comment:
This will call fetch on every isWalletConnected transition. Even when the wallet disconnects it will try to fetch.